libs/feature/search/src/lib/metadata-view/links/link-classifier.service.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core'
+import { MdViewFacade } from '../state/mdview.facade'
+import { map, startWith } from 'rxjs/operators'
+import { LinkHelperService } from './link-helper.service'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LinkClassifierService {
+  allLinks$ = this.facade.metadata$.pipe(
+    map((record) => (this.helper.hasLinks(record) ? record.links : []))
+  )
+  dataLinks$ = this.allLinks$.pipe(
+    map((links) => links.filter((link) => this.helper.isDataLink(link)))
+  )
+  otherLinks$ = this.allLinks$.pipe(
+    map((links) => links.filter((link) => this.helper.isOtherLink(link)))
+  )
+  validMapLinks$ = this.allLinks$.pipe(
+    map((links) => links.filter((link) => this.helper.isValidLink(link))),
+    map((links) => links.filter((link) => this.helper.isMapLink(link))),
+    startWith([])
+  )
```

Comment:
I realized that we now have two more services: LinkClassifier and LinkHelper. How about simply putting these observables in the facade? that would make sense IMO